plugins/modules/edpm_nova_derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,671 @@
+                specific to the NIC Partitioned PF
+            """
+            return DEL_USER_CONFIG
+    else:
+        """ The Product ID neither belongs to VF nor PF, simply ignore matching
+        """
+        return KEEP_USER_CONFIG
+
+
+# +------------+-----------------------+---------------------------+-------------------+
+# |   USER     |     product_id        |       product_id          |  Product ID       |
+# |   CONFIG   |     is VF             |       is PF               |  NOT mentioned    |
+# +------------+-----------------------+---------------------------+-------------------+
+# |   PCI addr |                       |                           |                   |
+# |   is VF    |     Matching VF       |       INVALID             |  add only VFs     |
+# |            |                       |                           |                   |
+# |            |                       |                           |                   |
+# +------------+-----------------------+---------------------------+-------------------+
+# |            |                       |                           |                   |
+# |   PCI addr |     ALL VFs of this   |       Matching PFs        | Add both PF and   |
+# |   is PF    |     addr -            |      - NIC Part PFs       | available VFs     |
+# |            |     NIC Part VFs      |                           |                   |
+# |            |                       |                           |                   |
+# +------------+-----------------------+---------------------------+-------------------+
+# |            |                       |                           |                   |
+# |  PCI addr  |     All matching      |       All matching        |  INVALID CASE     |
+# |     not    |     VFs               |       PFs - NIC Part PF   |                   |
+# |  specified |     NIC Part VFs      |                           |                   |
+# |            |                       |                           |                   |
+# +------------+-----------------------+---------------------------+-------------------+
+def get_passthrough_config(user_config, pf_name,
+                           allocated_pci, is_non_nic_pf: bool):
+    """Handle all variations of user specifid pci addr format
+
+    Arrive at the address fields of the whitelist.  Check the address fields of
+    the pci.passthrough_whitelist configuration option, validating the address fields.
+
+    :param user_configs: THT param NovaPCIPassthrough
+    :param pf_name: Interface/device name (str)
+    :param allocated_pci: List of VFs (for nic-partitioned PF), which are used by host
+    :param is_non_nic_pf: Indicates whether the PF is non-partitioned or not
+    :return: pci_passthrough: Derived config list
+    :return del_user_config: Flag to state if the user_config is to be deleted or not
+
+    Example format for user_config:
```

Comment:
https://github.com/openstack/nova/blob/master/nova/pci/devspec.py#L203-L211

     | [pci]
        | device_spec = {"address":"*:0a:00.*",
        |                "physical_network":"physnet1"}
        | device_spec = {"address": {"domain": ".*",
                                     "bus": "02",
                                     "slot": "01",
                                     "function": "[0-2]"},
                         "physical_network":"net1"}
        | device_spec = {"vendor_id":"1137","product_id":"0071"}

Reply:
Both "device_spec" and "passthrough_whitelist" will be supported - but the output conf file will have only "device_spec"